receiver/oracledbreceiver/scraper.go

```diff
@@ -430,6 +497,248 @@ func (s *oracleScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+		cacheUpdates++
+	}
+
+	// if cache updates is not equal to rows returned, that indicates there is problem somewhere
+	s.logger.Debug("Cache update", zap.Int("update-count", cacheUpdates), zap.Int("new-size", s.metricCache.Len()))
+
+	if len(hits) == 0 {
+		s.logger.Info("No log records for this scrape")
+		return plog.NewLogs(), errors.Join(errs...)
+	}
+
+	s.logger.Debug("Cache hits", zap.Int("hit-count", len(hits)), zap.Int("discarded-hit-count", discardedHits))
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Cache hit, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlID, hit.childNumber), zap.Int64("elapsed-time", hit.metrics[elapsedTimeMetric]))
+	}
+
+	// order by elapsed time delta, descending
+	sort.Slice(hits, func(i, j int) bool {
+		return hits[i].metrics[elapsedTimeMetric] > hits[j].metrics[elapsedTimeMetric]
+	})
+
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Cache hit after sorting, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlID, hit.childNumber), zap.String("child-address", hit.childAddress), zap.Int64("elapsed-time", hit.metrics[elapsedTimeMetric]))
+	}
+
+	// keep at most maxHitSize
+	hitCountBefore := len(hits)
+	maxHitsSize := min(len(hits), int(s.topQueryCollectCfg.TopQueryCount))
+	hits = hits[:maxHitsSize]
+	skippedCacheHits := hitCountBefore - len(hits)
+	s.logger.Debug("Skipped cache hits", zap.Int("count", skippedCacheHits))
+
+	for _, hit := range hits {
+		s.logger.Debug(fmt.Sprintf("Final cache hit, SQL_ID: %v, CHILD_NUMBER: %v", hit.sqlID, hit.childNumber), zap.String("child-address", hit.childAddress), zap.Any("metrics", hit.metrics))
+	}
+	hits = s.obfuscateCacheHits(hits)
+	childAddressToPlanMap := s.getChildAddressToPlanMap(ctx, hits)
+
+	rb := s.lb.NewResourceBuilder()
+	rb.SetOracledbInstanceName(s.instanceName)
+	rb.SetHostName(s.hostName)
+
+	for _, hit := range hits {
+		planBytes, err := json.Marshal(childAddressToPlanMap[hit.childAddress])
+		if err != nil {
+			s.logger.Error("Error marshaling plan data to JSON", zap.Error(err))
+		}
+		planString := string(planBytes)
+
+		s.lb.RecordDbServerTopQueryEvent(context.Background(),
+			timestamp,
+			dbSystemNameVal,
+			s.hostName,
+			hit.queryText,
+			planString, hit.sqlID, hit.childNumber,
+			asFloatInSeconds(hit.metrics[applicationWaitTimeMetric]),
+			hit.metrics[bufferGetsMetric],
+			asFloatInSeconds(hit.metrics[clusterWaitTimeMetric]),
+			asFloatInSeconds(hit.metrics[concurrencyWaitTimeMetric]),
+			asFloatInSeconds(hit.metrics[cpuTimeMetric]),
+			hit.metrics[queryDirectReadsMetric],
+			hit.metrics[queryDirectWritesMetric],
+			hit.metrics[queryDiskReadsMetric],
+			asFloatInSeconds(hit.metrics[elapsedTimeMetric]),
+			hit.metrics[queryExecutionMetric],
+			hit.metrics[physicalReadBytesMetric],
+			hit.metrics[physicalReadRequestsMetric],
+			hit.metrics[physicalWriteBytesMetric],
+			hit.metrics[physicalWriteRequestsMetric],
+			hit.metrics[rowsProcessedMetric],
+			asFloatInSeconds(hit.metrics[userIoWaitTimeMetric]))
```

Comment:
@sincejune this is pretty hard to read and error-prone when we have to many attribute. I guess events (maybe metrics as well?) might need a different interface to pass attribute as a struct e.g.:
```
s.lb.RecordDbServerTopQueryEvent(context.Background(), timestamp, internal.DbServerTopQueryEventAttributes{
	DBSystem: dbSystemNameVal,
	HostName: s.hostName,
	QueryText: hit.queryText,
	...
})
```
WDYT?